### DISCUSSION TITLE
VS Code doesn't offer gcc as a launch configuration

### DISCUSSION THREAD

**Discussion by matej-almasi** [microsoft/vscode-cpptools#11072] [Q&A]
6.1.31-2-MANJARO GNU/Linux
VS Code 1.79.0
C++ extension v1.15.4

Hello everyone. Switched to Manjaro Linux recently and tried to set up some basic c++ environment. I ran into a problem when I press the "play" button and get a prompt to pick a configuration, only "(gdb) launch" is offered to me:
![image](https://github.com/microsoft/vscode-cpptools/assets/120648152/0dbf6c21-ab50-4ce3-8998-32d6266faf3b)

I would expect a list of configurations, at least the ones related to gcc, I double checked if I have it installed.

Furthermore, when I select the offered "(gdb) launch)", no tasks.json is created, only "empty" launch.json and a following error message (of course - launch is empty):
![image](https://github.com/microsoft/vscode-cpptools/assets/120648152/f738cee5-0562-4a53-95b4-5c0b805633c1)
![image](https://github.com/microsoft/vscode-cpptools/assets/120648152/0104001c-4107-47e6-87cd-e8021a398391)

Does anyone know how to fix this? Could this be a distro issue?

By the way, I got the same behavior from both AUR and Snap installed VSC.

**Comment by sean-mcmanus** [comment]
@matej-almasi That means our extension isn't auto detecting a compiler at the expected locations. Do you know what the path is to the compiler you wanted to use? Is the compiler shown in the dropdown list when running the C/C++: Select IntelliSense Configuration... command?

**Reply by sean-mcmanus** [reply]
This could also be because you don't have a debugger installed (gdb or lldb). If you check your C/C++ logging you should see a message starting with "Unable to find the" if you're hitting that case (and you would also see the compiler in the C/C++: Select IntelliSense Configuration list since that doesn't exclude compilers without a debugger).

**Reply by matej-almasi** [reply]
Hello @sean-mcmanus , thanks for the reply.
It appears that both of the VS Code packages I installed were in a "sandbox" mode, preventing the application from properly interacting with the file system. As a Linux beginner, I didn't realize that. I checked the extension on a different distribution, making sure VSC isn't sandboxed. The extension works properly there.

Sorry for not providing this update earlier, I thougt there wouldn't be any replies after waiting a few days for some.
